ankimorphs/frequency_file_generator.py

```diff
@@ -43,20 +49,64 @@ def _setup_buttons(self) -> None:
         self.ui.createFrequencyFileButton.clicked.connect(self._generate_freqyency_file)
 
     def _on_input_button_clicked(self) -> None:
-        # TODO:
-        #  - we should ideally be able to select a directory OR a single file
-
-        tooltip("clicked select input button", parent=mw)
+        input_files = QFileDialog.getOpenFileNames(None, "Files to Analyze", "", "(*.txt *.srt)")
+        files = ""
+        if not input_files[0]:
+            return
+        for file_path in input_files[0]:
+            files += Path(file_path).name + " "
+        self.ui.lineEdit.setText(files)
+        self.path = Path(input_files[0][0]).parent
 
     def _on_output_button_clicked(self) -> None:
-        # TODO: should the user only be able to select a output
-        #   directory or should they be able to set the file name too?
-
-        tooltip("clicked select output button", parent=mw)
+        output_file = QFileDialog.getSaveFileName(None, "Save File", "", "(*.csv)")
+        print(output_file)
+        self.ui.lineEdit_2.setText(output_file[0])
 
     def _generate_freqyency_file(self) -> None:
-        # TODO:
-        #  - we need to use a QueryOp to run this in the background
-        #    like we do in recalc
+        field_content = self.ui.lineEdit.text()
+        text = self._read_files(field_content)
+        if not text:
+            return
+        am_config = AnkiMorphsConfig()
+        morphemizer = self._morphemizers[self.ui.comboBox.currentIndex()]
+        assert morphemizer is not None
+        morphs = get_morphemes(morphemizer, text, am_config)
+        frequency_list = self._generate_frequency_list(morphs)
+        if not self.ui.lineEdit_2.text():
+            tooltip("Output field is empty", parent=mw)
+            return
+        with open(self.ui.lineEdit_2.text(), mode='w+', encoding="utf-8", newline='') as csvfile:
+            spamwriter = csv.writer(csvfile)
+            for [inflected, base, _] in frequency_list:
+                spamwriter.writerow([inflected, base])
+
+    def _read_files(self, field_content: str) -> Optional[str]:
+        if field_content == "":
+            tooltip("Input field empty", parent=mw)
+            return None
+        if Path(field_content).is_file():
+            with open(field_content, mode="r", encoding="utf-8") as file:
+                return file.read()
+        else:
+            for file in field_content.split():
+                file_path = self.path.joinpath(file)
+                if not Path(file_path).is_file():
+                    tooltip(str(file_path) + " dosen't exist", parent=mw)
+                    return None
+                with open(file_path, mode="r", encoding="utf-8") as file:
+                    return file.read()
 
-        tooltip("clicked create frequency file button", parent=mw)
+    def _generate_frequency_list(self, morphes: list[Morpheme]) -> list[[str, str, int]]:
+        hash_map = {}
```

Comment:
a map means something else in python compared to most other programming languages (a map object is an iterator). This should be renamed to 'morph_occurrence_dict' or something to that affect.

Reply:
okay will get around to them

Reply:
@mortii I added the changes

Reply:
@Vilhelm-Ian ok, i'll test it.